sympy/polys/polytools.py

```diff
@@ -5652,6 +5652,9 @@ def _symbolic_factor_list(expr, opt, method):
             continue
         if arg.is_Pow:
             base, exp = arg.args
+            if base.is_Number and exp.is_Number:
```

Comment:
The more general thing to be doing (perhaps) is to use `if arg.is_number` to handle an expr like `sqrt(1+sqrt(2))`. (The base of that expression is a `number`, not a `Number`.